File: Examen_Unidad2/src/Mantenimiento.java
import java.text.SimpleDateFormat;
import java.util.*;

class Mantenimiento {
    private Empleado empleadoEncargado;
    private String idAnimal;
    private String procesoRealizado;
    private String fechaProceso;
    private String observaciones;

    public Mantenimiento(Empleado empleadoEncargado, String idAnimal, String procesoRealizado, String observaciones) {
        this.empleadoEncargado = empleadoEncargado;
        this.idAnimal = idAnimal;
        this.procesoRealizado = procesoRealizado;
        this.observaciones = observaciones;
        this.fechaProceso = obtenerFechaActual();
    }

    public Mantenimiento(String empleadoEncargado, String idAnimal, String procesoRealizado, String fechaMantenimiento) {
    }

    public Mantenimiento(String idAnimal, Empleado empleado, String procesoRealizado, String fechaMantenimiento, String observaciones) {
    }

    public Mantenimiento(String idAnimal, String empleadoEncargado, String procesoRealizado, String fechaMantenimiento, String observaciones) {
    }

    private String obtenerFechaActual() {
        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
        Date date = new Date();
        return formatter.format(date);
    }

    // Getters y Setters

    public Empleado getEmpleadoEncargado() {
        return empleadoEncargado;
    }

    public void setEmpleadoEncargado(Empleado empleadoEncargado) {
        this.empleadoEncargado = empleadoEncargado;
    }

    public String getIdAnimal() {
        return idAnimal;
    }

    public void setIdAnimal(String idAnimal) {
        this.idAnimal = idAnimal;
    }

    public String getProcesoRealizado() {
        return procesoRealizado;
    }

    public void setProcesoRealizado(String procesoRealizado) {
        this.procesoRealizado = procesoRealizado;
    }

    public String getFechaProceso() {
        return fechaProceso;
    }

    public void setFechaProceso(String fechaProceso) {
        this.fechaProceso = fechaProceso;
    }

    public String getObservaciones() {
        return observaciones;
    }

    public void setObservaciones(String observaciones) {
        this.observaciones = observaciones;
    }
}

File: Examen_Unidad2py/ZOO PY/Mantenimiento.py
from datetime import datetime

class Mantenimiento:
    def __init__(self, empleadoEncargado, idAnimal, procesoRealizado, observaciones):
        self.empleadoEncargado = empleadoEncargado
        self.idAnimal = idAnimal
        self.procesoRealizado = procesoRealizado
        self.observaciones = observaciones
        self.fechaProceso = self.obtenerFechaActual()

    def obtenerFechaActual(self):
        return datetime.now().strftime("%Y-%m-%d %H:%M:%S")

    # Getters and Setters
    def empleadoEncargado(self):
        return self._empleadoEncargado

    def empleadoEncargado(self, empleadoEncargado):
        self._empleadoEncargado = empleadoEncargado


    def idAnimal(self):
        return self._idAnimal

    @idAnimal.setter
    def idAnimal(self, idAnimal):
        self._idAnimal = idAnimal

    def procesoRealizado(self):
        return self._procesoRealizado

    def procesoRealizado(self, procesoRealizado):
        self._procesoRealizado = procesoRealizado

    def fechaProceso(self):
        return self._fechaProceso

    def fechaProceso(self, fechaProceso):
        self._fechaProceso = fechaProceso

    def observaciones(self):
        return self._observaciones

    def observaciones(self, observaciones):
        self._observaciones = observaciones